<?xml version="1.0"?>
<launch>
    <arg name="robot_model" default="$(optenv ROBOT_MODEL rbvogui)"/>
    <arg name="file_name" default="$(arg robot_model)_params.env"/>
    <arg name="file_path" default="/home/robot/robot_params/base/"/>
    <arg name="env_var" default="ROBOT_JOINT_POTENTIOMETER_VOLTAGE_CALIBRATION"/>
    <arg name="topic_sub" default="/robot/robotnik_base_hw/io"/>
    <arg name="calibration_duration" default="60"/>
    <arg name="calibrate_once" default="false"/>
    <arg name="save_to_file" default="true"/>
    <arg name="use_median" default="false"/>
    <arg name="restart_base_hw" default="true"/>
    <arg name="rosmon_node" default="rosmon_bringup"/>
    <arg name="robot_namespace" default="$(optenv ROBOT_ID robot)"/>
    <arg name="ros_param" default="/robot/robotnik_base_hw/joint_potentiometer_voltage_calibration"/>

    <node name="wheel_calibration" pkg="rbvogui_wheels_calibration" type="calibration_node.py" output="screen">
        <param name="file_name" value="$(arg file_name)"/>
        <param name="file_path" value="$(arg file_path)"/>
        <param name="env_var" value="$(arg env_var)"/>
        <param name="topic_sub" value="$(arg topic_sub)"/>
        <param name="calibration_duration" value="$(arg calibration_duration)"/>
        <param name="calibrate_once" value="$(arg calibrate_once)"/>
        <param name="save_to_file" value="$(arg save_to_file)"/>
        <param name="use_median" value="$(arg use_median)"/>
        <param name="restart_base_hw" value="$(arg restart_base_hw)"/>
        <param name="rosmon_node" value="$(arg rosmon_node)"/>
        <param name="robot_namespace" value="$(arg robot_namespace)"/>
        <param name="ros_param" value="$(arg ros_param)"/>
        <rosparam param="bringup_nodes_to_restart">
        - "robotnik_base_hw"
        - "controller_spawner"
        </rosparam>
    </node>
</launch>
